### ISSUE TITLE
Divide by zero

### ISSUE THREAD

**Issue by ntfrgl** [ntfrgl/test#41]
`core.network.Network.get_global_efficiency()`:

- 1 / `path_lengths`

`climate.climate_network.ClimateNetwork.get_correlation_distance_weighted_closeness()`:

- 1 / `self.get_correlation_distance()`

`climate.climate_network.ClimateNetwork.get_local_correlation_distance_weighted_vulnerability()`:

- 1 / `self.get_correlation_distance()`

----------------------------------------
Originally reported by: ntfrgl
Originally created on: 2014-06-17

**Comment by ntfrgl** [comment]
Ultimate reason for divide by zero in core.network.Network.get_global_efficiency() is that when recreating a network via an edge list, isolated nodes with degree zero are lost from the network:

```
#!python


edge_list = net_old.graph.get_edgelist() 
net_new = Network(edge_list=edge_list)
```

For example, if net_old contains one isolated node, this node will be lost in net_new. I.e., the number of nodes will be smaller: N_new = N_old - 1.

It seems that this issue can only ultimately be solved by introducing a vertex list additionally to the edge list, such as is the case in the igraph.Graph class.

An alternative solution could be to recreate networks via sparse adjacency matrices instead of edge lists in the core.network.Network.get_global_efficiency() method.

----------------------------------------
Original comment by: Jonathan Donges
Original comment on: 2014-09-25

**Comment by ntfrgl** [comment]
Solved via: *An alternative solution could be to recreate networks via sparse adjacency matrices instead of edge lists in the core.network.Network.get_global_efficiency() method.*

----------------------------------------
Original comment by: Jonathan Donges
Original comment on: 2014-09-25

**Comment by ntfrgl** [comment]
Resolved in commit 9775608 and commit 82c3628.

----------------------------------------
Original comment by: Jonathan Donges
Original comment on: 2014-09-26
